public/app/stores/device-list/device-list-store.js

```diff
@@ -0,0 +1,53 @@
+var Reflux = require('reflux'),
+  _ = require('lodash');
+
+var deviceActions = require('actions/device');
+var devicesService = require('servcies/devices/devices-service');
+var discoveryStore = require('stores/discovery-store');
+
+var deviceStore = Reflux.createStore({
+  init: function init () {
+    this.listenTo(discoveryStore, this.onDiscoveryUpdate);
+    this.listenTo(deviceActions.fetchList, this.onFetchList);
+
+    this.deviceList = {
+      model: [],
+      error: null
+    };
+    this.onFetchList();
+  },
+
+  getInitialState: function getInitialState () {
+    return this.deviceList;
+  },
+
+  onDiscoveryUpdate: function onDisoveryUpdate () {
+    this.onFetchList();
+  },
+
+  onFetchList: function onFetchList () {
+    var self = this;
+    devicesService.fetchDeviceList().subscribe(function onNext (next) {
```

Comment:
No need of anonymous function, you can use:
```
subscribe(self.updateModel.bind(self))
```

But there is even better option with ES6. 
If you create `updateModel` method using **fat arrow** `=>`, it will be bounded to `this` automatically and you can use it like that:
```
subscribe(self.updateModel)
```